tmt/base.py

```diff
@@ -545,20 +546,72 @@ def show(self):
         if self.opt('verbose'):
             self._sources()
 
+    def _lint_execute(self):
+        # execute is required for a plan
+        execute = self.node.get('execute')
+        if not execute:
+            echo(verdict(0, 'execute step must be defined with "how"'))
+            return False
+
+        how = execute.get('how')
+        if how not in ['detach', 'tmt']:
+            echo(verdict(0, f'unsupported execute method "{how}"'))
+            return False
+
+        return True
+
+    def _lint_summary(self):
+        # summary is advised with a resonable length
+        if self.summary is None:
+            echo(verdict(2, 'summary is very useful for quick inspection'))
+        elif len(self.summary) > 50:
+            echo(verdict(2, 'summary should not exceed 50 characters'))
+
+    def _lint_discover(self):
+        # discover step is optional
+        discover = self.node.get('discover')
+        if not discover:
+            return True
+
+        how = discover.get('how')
+        if how not in ['fmf', 'shell']:
+            echo(verdict(0, f'unknown discover method "{how}"'))
+            return False
+
+        return self._lint_discover_fmf(discover)
+
+    @staticmethod
+    def _lint_discover_fmf(discover):
+        # default to `None` for non-existing keys
+        fmf_id = collections.defaultdict(lambda: None, {
```

Comment:
Is this necessary? Shouldn't  https://github.com/psss/tmt/blob/d737f95597b13ff2f960b279c96e71808ee3bbad/tmt/base.py#L104 be used to get fmf_id?

Reply:
I will take a look, maybe I misunderstood what that function does

Reply:
I'm not sure this will work as `discover` step data are not a `Node` instance, see [modules](https://tmt.readthedocs.io/en/latest/modules.html) overview.

Reply:
yep, that was my understanding also. So keeping it as it is ...